tiledb/api/examples/multi_range_subarray.rs

```diff
@@ -0,0 +1,100 @@
+extern crate tiledb;
+
+use itertools::izip;
+use serde_json::json;
+
+use tiledb::array::{
+    Array, ArrayType, AttributeData, CellOrder, DimensionData, DomainData,
+    SchemaData, TileOrder,
+};
+use tiledb::context::Context;
+use tiledb::query::{
+    Query, QueryBuilder, ReadBuilder, ReadQuery, ReadQueryBuilder, WriteBuilder,
+};
+use tiledb::Result as TileDBResult;
+use tiledb::{Datatype, Factory};
+
+const ARRAY_URI: &str = "multi_range_slicing";
+
+fn main() -> TileDBResult<()> {
+    let ctx = Context::new()?;
+    if !Array::exists(&ctx, ARRAY_URI)? {
+        create_array(&ctx)?;
+        write_array(&ctx)?;
+    }
+
+    let array = Array::open(&ctx, ARRAY_URI, tiledb::array::Mode::Read)?;
+    let mut query = ReadBuilder::new(array)?
+        .layout(tiledb::query::QueryLayout::RowMajor)?
+        .register_constructor::<_, Vec<i32>>("rows", Default::default())?
+        .register_constructor::<_, Vec<i32>>("cols", Default::default())?
+        .register_constructor::<_, Vec<i32>>("a", Default::default())?
+        .start_subarray()?
+        .add_range("rows", &[1, 2])?
+        .add_range("rows", &[4, 4])?
+        .add_range("cols", &[1, 4])?
+        .finish_subarray()?
+        .build();
+
+    let (a, (cols, (rows, ()))) = query.execute()?;
+
+    for (row, col, a) in izip!(rows, cols, a) {
+        println!("{} {} {}", row, col, a);
```

Comment:
What is the expected result supposed to be? Can you add comments?

Reply:
[Fix example description](https://github.com/TileDB-Inc/tiledb-rs/pull/72/commits/85e01c0022fb3ebf0acfaea3af0ba4c405edcc74)